### PR TITLE
feat: avoid pulling unnecessary dependencies for `enhanced-determinism`

### PR DESCRIPTION
**Pull request by Aceeri**
Just fixin' up what I started.

https://github.com/jcornaz/heron/pull/180
https://blog.rust-lang.org/2022/04/07/Rust-1.60.0.html#new-syntax-for-cargo-features

### CODE REVIEW

**Comment by jcornaz** [comment]
Awesome. That is one of the rust features I was most anticipating. I am so glad that it is now stable :-)

I'll try to merge this soon ;-)

**Review by jcornaz — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -101,6 +101,8 @@ jobs:
           toolchain: nightly
           override: true
           profile: minimal
+      - run: cargo --version
+      - run: rustc --version
```
[the remark this answers is not in the document]
Comment (Aceeri) [reply .github/workflows/build.yml:105]: Yeah, was just trying to figure out what was going on with the failing unused-dependencies test.

[the remark this answers is not in the document]
Comment (jcornaz) [reply .github/workflows/build.yml:105]: it looks like a bug of the nightly compiler to me. As the build passes on the stable channel and fails only for that job which is using the nightly channel.

[the remark this answers is not in the document]
Comment (jcornaz) [reply .github/workflows/build.yml:105]: Or, maybe a bug of `cargo-udeps`

[the remark this answers is not in the document]
Comment (jcornaz) [reply .github/workflows/build.yml:105]: https://github.com/est31/cargo-udeps/issues/116

[the remark this answers is not in the document]
Comment (jcornaz) [reply .github/workflows/build.yml:105]: It has been solved by `cargo-udeps`

Comment (jcornaz) [inline .github/workflows/build.yml:105]:
```suggestion
      - run: cargo --version
      - run: rustc --version
```

**Comment by jcornaz** [comment]
As this requires the latest version of rust, and a new major version of bevy should come out this month, I will wait to update to bevy 0.7 first, and include it in heron 3.0.0.

**Pull request merged**